### DISCUSSION TITLE
Optional parameters using TeamCity

### DISCUSSION THREAD

**Discussion by DTTerastar** [cake-build/cake#3104] [Q&A] [answered]
When we pass in an optional parameter with teamcity and we haven't filled in a parameter it does it via --option=

I see that on https://cakebuild.net/docs/getting-started/upgrade#passing-empty-arguments this was done on purpose.

It now seems all optional parameters are impossible using one of the mainstream CI/CD servers.  Any idea on how to accomplish this now on 1.0?

**Comment by DTTerastar** [comment]
One idea is we could regex out the = since it works without it, or if there is any value.  So have the powershell bootstrapper regex out the "bad" arguments.

**Comment by augustoproiete** [comment, marked as answer]
@DTTerastar The answers on this question "[How do you pass empty arguments in Cake v1?](https://stackoverflow.com/a/66125812/211672)" on StackOverflow might be what you're looking for.

**Reply by pascalberger** [reply]
`--option %TC_VARIABLE%` should work with 1.0.

We also updated [upgrade instruction to mention this](https://cakebuild.net/docs/getting-started/upgrade#passing-empty-arguments).

**Reply by augustoproiete** [reply]
I believe you mean `--option %TC_VARIABLE%` without the `=`

**Reply by pascalberger** [reply]
Sure, fixed it
